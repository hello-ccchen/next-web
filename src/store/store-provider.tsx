"use client";
import { Provider } from "react-redux";
import { CommonProps } from "@/interfaces/common/common-interface";
import { store } from "./store";

function StoreProvider({ children }: CommonProps) {
  return <Provider store={store}>{children}</Provider>;
}

export default StoreProvider;
