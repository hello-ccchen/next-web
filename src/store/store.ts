import { configureStore } from "@reduxjs/toolkit";
import { TypedUseSelectorHook, useDispatch, useSelector } from "react-redux";
import usersReducer from "@/store/users/users-slice";
import productsReducer from "@/store/products/products-slice";

export const store = configureStore({
  reducer: {
    users: usersReducer,
    products: productsReducer
  },
});

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

// Use throughout your app instead of plain `useDispatch` and `useSelector`
export const useAppDispatch: () => AppDispatch = useDispatch;
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
