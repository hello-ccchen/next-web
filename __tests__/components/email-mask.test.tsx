/**
 * @jest-environment jsdom
 */
import { fireEvent, render } from "@testing-library/react";
import EmailMask from "@/components/users/email-mask";
import * as reactRedux from "@/store/store";
import { mockFilteredUserResponseDataWithOnePage } from "../config/mocks/users-data.mock";

describe("EmailMask component", () => {
  const useAppDispatchMock = jest.spyOn(reactRedux, "useAppDispatch");

  afterEach(() => {
    useAppDispatchMock.mockClear();
  });

  it("email should masked correctly", () => {
    useAppDispatchMock.mockReturnValue(jest.fn());
    const { container, getByText } = render(<EmailMask {...mockFilteredUserResponseDataWithOnePage[0]} />);
    expect(container).toMatchSnapshot();
    expect(getByText('*****')).toBeTruthy();
  });

  it("should reveal email when revealEmail button is clicked", () => {
    useAppDispatchMock.mockReturnValue(jest.fn());
    const { getByAltText } = render(<EmailMask {...mockFilteredUserResponseDataWithOnePage[0]} />);
    const revealEmailButton = getByAltText('mask');
    fireEvent.click(revealEmailButton);
    expect(useAppDispatchMock).toHaveBeenCalled();
  });
});
