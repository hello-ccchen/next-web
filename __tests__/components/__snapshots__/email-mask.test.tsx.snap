// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailMask component email should masked correctly 1`] = `
<div>
  <div
    class="d-flex align-items-center"
  >
    <span>
      *****
    </span>
    <img
      alt="mask"
      class="ms-2"
      data-nimg="1"
      decoding="async"
      height="16"
      loading="lazy"
      src="/eye.svg"
      style="color: transparent; cursor: pointer;"
      width="16"
    />
  </div>
</div>
`;
